src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -381,47 +394,132 @@ private void DropItems(Character killer)
 
-				if (killer == null || dropChance > autoloot)
-				{
-					var direction = new Direction(rnd.Next(0, 360));
-					var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
-					var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+			if (rnd.NextDouble() * 100D < worldConf.GoldJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.TwinkleBuff, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
 
-					dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
-					dropItem.Drop(this.Map, this.Position, direction, distance);
-				}
-				else
-				{
-					killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
-				}
+			if (rnd.NextDouble() * 100D < worldConf.BlueJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperExp, 1, 0, TimeSpan.Zero, this);
+				return;
 			}
+
+			if (rnd.NextDouble() * 100D < worldConf.SilverJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperDrop, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
+
+			// Check to see if this monster can become elite
+			// Note that on official it uses the map level rather than the monster level
+			if (this.Level < worldConf.EliteMinLevel)
+				return;
+
+			if (redOrb)
+				spawnRateBoost = worldConf.RedOrbEliteRate / 100F;
+
+			if (rnd.NextDouble() * 100D < worldConf.EliteSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.EliteMonsterBuff, 1, 0, TimeSpan.Zero, this);
+
+				// Increase its stats
+
+				var properties = this.Properties;
+				var overrides = this.Properties.Overrides;
+
+				overrides.SetFloat(PropertyName.MHP, properties.GetFloat(PropertyName.MHP) * worldConf.EliteHPSPRate / 100F);
+				overrides.SetFloat(PropertyName.MSP, properties.GetFloat(PropertyName.MSP) * worldConf.EliteHPSPRate / 100F);
+
+				overrides.SetFloat(PropertyName.MINPATK, properties.GetFloat(PropertyName.MINPATK) * worldConf.EliteStatRate / 100F);
+				overrides.SetFloat(PropertyName.MAXPATK, properties.GetFloat(PropertyName.MAXPATK) * worldConf.EliteStatRate / 100F);
+				overrides.SetFloat(PropertyName.MINMATK, properties.GetFloat(PropertyName.MINMATK) * worldConf.EliteStatRate / 100F);
+				overrides.SetFloat(PropertyName.MAXMATK, properties.GetFloat(PropertyName.MAXMATK) * worldConf.EliteStatRate / 100F);
+				overrides.SetFloat(PropertyName.DEF, properties.GetFloat(PropertyName.DEF) * worldConf.EliteStatRate / 100F);
+				overrides.SetFloat(PropertyName.MDEF, properties.GetFloat(PropertyName.MDEF) * worldConf.EliteStatRate / 100F);
+
+				properties.InvalidateAll();
+
+				properties.SetFloat(PropertyName.HP, properties.GetFloat(PropertyName.MHP));
+				properties.SetFloat(PropertyName.SP, properties.GetFloat(PropertyName.MSP));
+
+				if (worldConf.EliteAlwaysAggressive) this.Tendency = TendencyType.Aggressive;
+
+				// TODO: Also needs to be able to summon its minions and have its special attack
+
+				return;
```

Comment:
This appears to be unnecessary.

Reply:
The return statement?  You're right about that, I removed it.